nbbuild/misc/prepare-bundles/src/main/java/org/netbeans/prepare/bundles/PrepareBundles.java

```diff
@@ -0,0 +1,295 @@
+                }
+            }
+            for (String prj : use.projects) {
+                project2LicenseKey.put(prj, licenseName);
+            }
+        }
+        for (Entry<String, LicenseDescription> e : project2License.entrySet()) {
+            LicenseDescription licenseDesc = e.getValue();
+            Path projectLicenseFile = bundlesDir.resolve(e.getKey() + "-" + licenseDesc.version + "-license.txt");
+            try (Writer w = new OutputStreamWriter(Files.newOutputStream(projectLicenseFile), "UTF-8")) {
+                w.write("Name: " + licenseDesc.name + nl);
+                w.write("Description: " + licenseDesc.description + nl);
+                w.write("Version: " + licenseDesc.version + nl);
+                w.write("License: " + project2LicenseKey.get(e.getKey()) + nl);
+                w.write("Origin: " + licenseDesc.homepage + nl);
+                w.write(nl);
+                w.write(licenseDesc.licenseText);
+            }
+            String notice = project2Notice.get(e.getKey());
+            if (notice != null) {
+                Path projectNoticeFile = bundlesDir.resolve(e.getKey() + "-" + licenseDesc.version + "-notice.txt");
+                try (Writer w = new OutputStreamWriter(Files.newOutputStream(projectNoticeFile), "UTF-8")) {
+                    w.write(notice);
+                }
+            }
+        }
+    }
+
+    private static String readString(Path p) throws IOException {
+        return new String(Files.readAllBytes(p), StandardCharsets.UTF_8);
+    }
+
+    private static List<String> licenseTextToTokens(String licenseText) {
+        return Arrays.asList(licenseText.replaceAll("[ \n\r\t]+", " ").split(" "));
+    }
+    private static class LicenseDescription {
+        private final String name;
+        private final String version;
+        private final String description;
+        private final String homepage;
+        private final String licenseKey;
+        private final String licenseText;
+        private final List<String> bundles = new ArrayList<>();
+
+        public LicenseDescription(String name, String version, String description, String homepage, String licenseKey, String licenseText) {
+            this.name = name;
+            this.version = version;
+            this.description = description;
+            this.homepage = homepage;
+            this.licenseKey = licenseKey;
+            this.licenseText = licenseText;
+        }
+
+        @Override
+        public String toString() {
+            return "LicenseDescription{" + "version=" + version + ", description=" + description + ", homepage=" + homepage + ", licenseKey=" + licenseKey + ", licenseText=" + licenseText + ", bundles=" + bundles + '}';
+        }
+
+    }
+    private static class LicenseUses {
+        private final String key;
+        private final String licenseText;
+        private final List<String> projects = new ArrayList<>();
```

Comment:
Accessing a private field from outside the class looks strange to me. Would you consider moving the declarations here to `public final`?